keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -547,57 +497,68 @@ This includes 3 scenarios:
 For each exported service, one `ServiceExport` will exist in each cluster that
 runs the service. The mcsd-controller will create and maintain a derived
 `ServiceImport` in each cluster within the supercluster (see: [constraints and
-conflict resolution](#constraints-and-conflict-resolution)). If all `ServiceExport` instances are deleted, each
-`ServiceImport` will also be deleted from all clusters.
+conflict resolution](#constraints-and-conflict-resolution)). If all
+`ServiceExport` instances are deleted, each `ServiceImport` will also be deleted
+from all clusters.
 
 Since a given `ServiceImport` may be backed by multiple `EndpointSlices`, a
 given `EndpointSlice` will reference its `ServiceImport` using the label
 `multicluster.kubernetes.io/imported-service-name` similarly to how an
 `EndpointSlice` is associated with its `Service` in a single cluster. Each
 imported `EndpointSlice` will also have a
-`multicluster.kubernetes.io/source-cluster` label with a registry-scoped unique
-identifier for the cluster.
+`multicluster.kubernetes.io/source-cluster` label with the cluster name, a
+registry-scoped unique identifier for the cluster.
 
 ```golang
-// ServiceImport declares that the specified service should be exported to other clusters.
+// ServiceImport describes a service imported from clusters in a supercluster.
 type ServiceImport struct {
- metav1.TypeMeta `json:",inline"`
- // +optional
- metav1.ObjectMeta `json:"metadata,omitempty"`
- // +optional
- Spec ServiceImportSpec `json:"spec,omitempty"`
+  metav1.TypeMeta `json:",inline"`
+  // +optional
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+  // +optional
+  Spec ServiceImportSpec `json:"spec,omitempty"`
+  // +optional
+  Status ServiceImportStatus `json:"status,omitempty"`
 }
 
-// ServiceImportSpec contains the current status of an imported service and the
-// information necessary to consume it
+// ServiceImportType designates the type of a ServiceImport
+type ServiceImportType string
+
+const (
+  // SuperclusterIP are only accessible via the Supercluster IP.
+  SuperclusterIP ServiceImportType = "SuperclusterIP"
+  // Headless services allow backend pods to be addressed directly.
+  Headless ServiceImportType = "Headless"
+)
+
+// ServiceImportSpec describes an imported service and the information necessary to consume it.
 type ServiceImportSpec struct {
   // +patchStrategy=merge
```

Comment:
We need to think about this API a bit - Service Ports have been no end of pain because the merge key wasn't unique enough

Reply:
Maybe we should just make this an atomic list?  Or make the key be the name, but I am not sure "" counts as a valid merge key

Reply:
Let's plan a formal API review

Reply:
Yes to the API review.

I think an atomic list makes sense here - early drafts used one. I don't think there's a strong reason to support merge for service ports given how we expect service ports to be updated.